src/ion.hpp

```diff
@@ -71,19 +74,50 @@ public :
                +  sizeof(ion);
     }
 
+    void set_valency(int v) {
+        valency_ = v;
+    }
+    int valency() const {
+        return valency_;
+    }
+
+    void set_default_internal_concentration(value_type c) {
+        EXPECTS(c>=value_type(0));
+        Xi_default_ = c;
+    }
+    void set_default_external_concentration(value_type c) {
+        EXPECTS(c>=value_type(0));
+        Xo_default_ = c;
+    }
+
     view current() {
         return iX_;
     }
+
     view reversal_potential() {
         return eX_;
     }
+
     view internal_concentration() {
         return Xi_;
     }
+
     view external_concentration() {
         return Xo_;
     }
 
+    void reset() {
+        memory::fill(iX_, 0);
+        memory::fill(Xi_, Xi_default_);
+        memory::fill(Xo_, Xo_default_);
+    }
+
+    /// Calculate the reversal potential for all compartments using Nernst equation
+    /// temperature is in degrees Kelvin
+    void update_reversal_potential(value_type temperature) {
```

Comment:
I think the member function name should indicate that it updates the reversal potential according to the Nernst equation.

Reply:
done.